### PR TITLE
chore(integer): add crt 32 bits tests with 5_1 params

### PR DESCRIPTION
**Pull request by IceTDrinker**
### PR content/description

- remove buggy unchecked_scalar_add_assign and replace by the proper implementation which had a different name

BREAKING CHANGE:
removed an API entry point which was not required

### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
sick of those nvm crashes -_-

**Comment by IceTDrinker** [comment]
@slab-ci cpu_fast_test

File: tfhe/src/integer/server_key/crt/tests.rs [hunk 1]
```diff
@@ -65,6 +97,163 @@ fn integer_crt_mul_32_bits() {
...
+
+#[test]
+fn integer_unchecked_crt_sub_32_bits() {
+    let params = PARAM_MESSAGE_5_CARRY_1_KS_PBS;
+
+    // Define CRT basis, and global modulus
+    let basis = [3u64, 11, 13, 19, 23, 29, 31, 32];
+
+    // Use u128 to avoid overflows as the modulus is slightly larger than 32 bits
+    let modulus = basis.iter().map(|x| *x as u128).product::<u128>();
```
Comment (tmontaigu) [inline tfhe/src/integer/server_key/crt/tests.rs:136]: minor thing: `iter().copied.produc::<u128>()`

File: tfhe/src/integer/server_key/crt/scalar_sub_crt.rs [hunk 2]
```diff
@@ -42,8 +42,7 @@ impl ServerKey {
         //Put each decomposition into a new ciphertext
         for (ct_i, mod_i) in ct.blocks.iter_mut().zip(ct.moduli.iter()) {
             let neg_scalar = (mod_i - scalar % mod_i) % mod_i;
-            self.key
-                .unchecked_scalar_add_assign_crt(ct_i, neg_scalar as u8);
+            self.key.unchecked_scalar_add_assign(ct_i, neg_scalar as u8);
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/server_key/crt/scalar_sub_crt.rs:45]: must have missed that afterward but that's a good point

File: tfhe/src/integer/server_key/crt/tests.rs (hunk 1 above)
Comment (IceTDrinker) [reply tfhe/src/integer/server_key/crt/tests.rs:136]: indeed I think I took that code from somewhere else

File: tfhe/src/integer/server_key/crt/scalar_sub_crt.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/server_key/crt/scalar_sub_crt.rs:45]: actually that function is already remove I believe :)

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/integer/server_key/crt/scalar_sub_crt.rs:45]: only traces left of CRT in shortint are encryption which may need their own visit

File: tfhe/src/integer/server_key/crt/tests.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/integer/server_key/crt/tests.rs]: Ok

File: tfhe/src/integer/server_key/crt/scalar_sub_crt.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tmontaigu) [reply tfhe/src/integer/server_key/crt/scalar_sub_crt.rs:45]: Yes indeed it was removed in this pr, just missed it in the diff

**Review by tmontaigu — APPROVED** [review]

**Comment by IceTDrinker** [comment]
will check https://github.com/zama-ai/tfhe-rs-internal/issues/256 before merging as it may be related/fixed/fixable

**Comment by IceTDrinker** [comment]
looks unrelated, will merge this and push the fix in a separate PR

**Pull request merged**